homeassistant/components/homematic/cover.py

```diff
@@ -49,7 +49,9 @@ def current_cover_position(self) -> int | None:
 
         None is unknown, 0 is closed, 100 is fully open.
         """
-        return int(self._hm_get_state() * 100)
+        state = self._hm_get_state()
+        if isinstance(state, int):
+            return int(state * 100)
```

Comment:
Added a `return None` - there's codebases that just don't use any explicit return in such cases.

I'm not sure on the try/except.
I also didn't dig too deep - what would actually be the expected return of `_hm_get_state()` for an "unknown" state? Usually I'd expect `None`, but at least in my setup it returns an empty string.

If the empty string is the proper value to expect (and not e.g. `None`) one could also check for `state == ''` or even introduce a constant like `STATE_UNKOWN = ''`.

Reply:
Another thought, are you sure the result of `self._hm_get_state()` is of the type `int`?
The original code creates an `int` to return. So I'm assuming it's not?
I checked the function in `entity.py`, it's not typed so no info there.
It could be a `str`, please double check.

Reply:
Good point. It's of course a float between 0 and 1 - added some `print` statements to my locally running instance.
A `str` would not have worked with the `int(value*100)`.